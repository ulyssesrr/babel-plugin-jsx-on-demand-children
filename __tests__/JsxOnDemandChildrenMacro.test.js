'use strict';

const plugin = require('babel-plugin-macros');
const pluginTester = require('babel-plugin-tester').default

pluginTester({
  plugin,
  snapshot: true,
  title: 'JsxOnDemandChildrenMacro',
  babelOptions: { filename: __filename, parserOpts: { plugins: ['jsx'] } },
  tests: {
    works: {
      code: `
        'use strict';

        import React from 'react';
        import JsxOnDemandChildren from '../src/macro';

        function Test() {
          return (
            <Parent>
              <JsxOnDemandChildren>
                <Parent.LeafChild prop={true} />
              </JsxOnDemandChildren>
            </Parent>
          );
        };
      `,
    },
    'Invalid usage: no parent': {
      code:`
        'use strict';

        import React from 'react';
        import JsxOnDemandChildren from '../src/macro';

        JsxOnDemandChildren;

        function Test() {
          return (
            <Parent>
              <JsxOnDemandChildren>
                <Parent.LeafChild prop={true} />
              </JsxOnDemandChildren>
            </Parent>
          );
        };
      `,
      error: SyntaxError,
    },
    'Invalid usage: not JSXElement': {
      code: `
        'use strict';

        import React from 'react';
        import JsxOnDemandChildren from '../src/macro';

        JsxOnDemandChildren.invalid = true;

        function Test() {
          return (
            <Parent>
              <JsxOnDemandChildren>
                <Parent.LeafChild prop={true} />
              </JsxOnDemandChildren>
            </Parent>
          );
        };
      `,
      error: SyntaxError,
    },
  },
});