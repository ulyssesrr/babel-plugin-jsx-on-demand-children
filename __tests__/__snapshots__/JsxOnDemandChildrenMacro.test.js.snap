// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JsxOnDemandChildrenMacro Invalid usage: no parent: Invalid usage: no parent 1`] = `

'use strict';

import React from 'react';
import JsxOnDemandChildren from '../src/macro';

JsxOnDemandChildren;

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <Parent.LeafChild prop={true} />
      </JsxOnDemandChildren>
    </Parent>
  );
};

      â†“ â†“ â†“ â†“ â†“ â†“

SyntaxError: /home/ulysses/git/personal/react/babel-plugin-jsx-on-demand-children/__tests__/JsxOnDemandChildrenMacro.test.js: ../src/macro: Invalid usage of JsxOnDemandChildren.
[0m [90m 4 |[39m [36mimport[39m [33mJsxOnDemandChildren[39m [36mfrom[39m [32m'../src/macro'[39m[33m;[39m[0m
[0m [90m 5 |[39m[0m
[0m[31m[1m>[22m[39m[90m 6 |[39m [33mJsxOnDemandChildren[39m[33m;[39m[0m
[0m [90m   |[39m [31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[0m
[0m [90m 7 |[39m[0m
[0m [90m 8 |[39m [36mfunction[39m [33mTest[39m() {[0m
[0m [90m 9 |[39m   [36mreturn[39m ([0m

`;

exports[`JsxOnDemandChildrenMacro Invalid usage: not JSXElement: Invalid usage: not JSXElement 1`] = `

'use strict';

import React from 'react';
import JsxOnDemandChildren from '../src/macro';

JsxOnDemandChildren.invalid = true;

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <Parent.LeafChild prop={true} />
      </JsxOnDemandChildren>
    </Parent>
  );
};

      â†“ â†“ â†“ â†“ â†“ â†“

SyntaxError: /home/ulysses/git/personal/react/babel-plugin-jsx-on-demand-children/__tests__/JsxOnDemandChildrenMacro.test.js: ../src/macro: Invalid usage of JsxOnDemandChildren.
[0m [90m 4 |[39m [36mimport[39m [33mJsxOnDemandChildren[39m [36mfrom[39m [32m'../src/macro'[39m[33m;[39m[0m
[0m [90m 5 |[39m[0m
[0m[31m[1m>[22m[39m[90m 6 |[39m [33mJsxOnDemandChildren[39m[33m.[39minvalid [33m=[39m [36mtrue[39m[33m;[39m[0m
[0m [90m   |[39m [31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[31m[1m^[22m[39m[0m
[0m [90m 7 |[39m[0m
[0m [90m 8 |[39m [36mfunction[39m [33mTest[39m() {[0m
[0m [90m 9 |[39m   [36mreturn[39m ([0m

`;

exports[`JsxOnDemandChildrenMacro works: works 1`] = `

'use strict';

import React from 'react';
import JsxOnDemandChildren from '../src/macro';

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <Parent.LeafChild prop={true} />
      </JsxOnDemandChildren>
    </Parent>
  );
};

      â†“ â†“ â†“ â†“ â†“ â†“

"use strict";

import React from "react";

function Test() {
  return <Parent LeafChild={() => <Parent.LeafChild prop={true} />}></Parent>;
}


`;
